### DISCUSSION TITLE
CLIFM crashes with segmentation fault.

### DISCUSSION THREAD

**Discussion by anoduck** [leo-arch/clifm#254] [Q&A] [answered]
A few weeks ago, CLIFM began crashing with a segmentation fault error, and I thought I had already opened an issue for this, but was mistaken. It was not crashing in GDB, but now is. This began a little over three weeks ago.

**Comment by leo-arch** [comment]
Hey @anoduck, thanks for reporting! Could you provide some more info about this issue? When exactly does it crash? terminal, OS, clifm version, install procedure, command line parameters (if any), etc?

**Reply by anoduck** [reply]
Term: Kitty-xterm
OS: OpenBSD-Snapshot
Shell: Zsh with ZSHELL integration (It is so hot)
CLIFM Version: 1.14.3
Exec: `clifm`
Install Precd: `sudo make install clean` as per instructions.

It crashes on start, and I have already wiped the configuration clean. So it should be on default. I can also copy and paste my `.zshrc` if needed. 

Here is the debug out put from GDB:

```gdb
Starting program: /usr/bin/clifm 

Program received signal SIGSEGV, Segmentation fault.
0x000004e90130f139 in memcpy (dst0=0x4e8f0dda8c0, src0=<optimized out>, length=280) at /usr/src/lib/libc/string/memcpy.c:103
103		TLOOP(*(word *)dst = *(word *)src; src += wsize; dst += wsize);
(gdb) bt
#0  0x000004e90130f139 in memcpy (dst0=0x4e8f0dda8c0, src0=<optimized out>, length=280) at /usr/src/lib/libc/string/memcpy.c:103
#1  0x000004e6bb0d6963 in x_scandir ()
#2  0x000004e6bb0f276e in get_path_programs ()
#3  0x000004e6bb105e25 in main ()
(gdb) info args
dst0 = 0x4e8f0dda8c0
src0 = <optimized out>
length = 280
(gdb) info locals
dst = <optimized out>
src = 0x4e975120f10 "\035\017^"
t = 35
(gdb) info frame
Stack level 0, frame at 0x797ee0f099a0:
 rip = 0x3d5e5fc9139 in memcpy (/usr/src/lib/libc/string/memcpy.c:103); saved rip = 0x3d30534f963
 called by frame at 0x797ee0f09a10
 source language c.
 Arglist at 0x797ee0f09568, args: dst0=0x3d5dff66b40, src0=<optimized out>, length=280
 Locals at 0x797ee0f09568, Previous frame's sp is 0x797ee0f099a0
 Saved registers:
  rbx at 0x797ee0f09978, rbp at 0x797ee0f09990, r11 at 0x797ee0f09988, r14 at 0x797ee0f09980, rip at 0x797ee0f09998
```

**Reply by leo-arch** [reply]
Ok, looks promising. I wonder why on earth is it calling x_scandir()? Don't worry, just thinking aloud.

**Reply by leo-arch** [reply, marked as answer]
Give it a try now @anoduck.

**Reply by anoduck** [reply]
HA! It Works! We are back in business!
